src/gpu/intel/jit/ir/kernel_info.hpp

```diff
@@ -116,19 +116,23 @@ class kernel_info_t {
 
     const compute::nd_range_t &nd_range() const { return nd_range_; }
 
-    void register_internal_arg(
+    void register_immediate_arg(
             const expr_t &var, const expr_t &value = expr_t()) {
         register_arg(
-                var, arg_kind_t::internal, DNNL_ARG_UNDEF, /*is_input=*/true);
-        set_internal_arg(var.as<var_t>().name, value);
+                var, arg_kind_t::immediate, DNNL_ARG_UNDEF, /*is_input=*/true);
+        set_immediate_arg(var.as<var_t>().name, value);
     }
 
-    void set_internal_arg(const std::string &name, const expr_t &value) {
+    void set_immediate_arg(const std::string &name, const expr_t &value) {
         auto *arg = find_arg_impl(name);
         gpu_assert(arg) << "Cannot find argument: " << name;
         arg->value = value;
     }
 
+    void register_user_hostscalar_arg(const expr_t &var, int dnnl_arg) {
```

Comment:
To align naming with public API naming:
```suggestion
    void register_user_host_scalar_arg(const expr_t &var, int dnnl_arg) {
```